### PR TITLE
Rebalancing Set Bid with ETH

### PR DESCRIPTION
**Pull request by richardliang**
- Future PR will incorporate getting the BidPlacedWithEth event emitted from this contract

### CODE REVIEW

File: test/helpers/rebalancingSetBidderHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const contract = require('truffle-contract');
+
+export const deployRebalancingSetEthBidderAsync = async (
+  web3: Web3,
+  rebalanceAuctionModule: RebalanceAuctionModuleContract,
+  transferProxy: TransferProxyContract,
+  wrappedEther: WethMockContract,
+  owner: Address = DEFAULT_ACCOUNT,
+): Promise<RebalancingSetEthBidderContract> => {
+  const truffleRebalancingSetEthBidderContract = contract(RebalancingSetEthBidder);
```
Comment (felix2feng) [inline test/helpers/rebalancingSetBidderHelpers.ts]: We might have a helper that abstracts all these steps away.

`setDefaultTruffleContract` in coreHelpers. If you have time, would be good to refactor all the Helpers to use this

Comment (richardliang) [reply test/helpers/rebalancingSetBidderHelpers.ts]: Done

File: src/api/RebalancingAPI.ts [hunk 2]
```diff
@@ -577,6 +611,44 @@ export class RebalancingAPI {
...
+  private async assertBidWithEther(rebalancingSetTokenAddress: Address, bidQuantity: BigNumber, txOpts: Tx) {
+    this.assert.schema.isValidAddress('rebalancingSetTokenAddress', rebalancingSetTokenAddress);
+    this.assert.common.greaterThanZero(
+      bidQuantity,
+      coreAPIErrors.QUANTITY_NEEDS_TO_BE_POSITIVE(bidQuantity)
+    );
+
+    await this.assert.setToken.isValidSetToken(this.core.coreAddress, rebalancingSetTokenAddress);
+    await this.assert.rebalancing.isInRebalanceState(rebalancingSetTokenAddress);
+    await this.assert.rebalancing.bidAmountLessThanRemainingSets(rebalancingSetTokenAddress, bidQuantity);
```
Comment (bweick) [inline src/api/RebalancingAPI.ts]: Feels like this assertion should be conditioned on if they're allowing a partialFill or not

Comment (richardliang) [reply src/api/RebalancingAPI.ts]: Yeah agree this is more accurate. Will also update in the bidAsync assertion

**Pull request merged**